docs/book/src/tasks/upgrading-cluster-api-versions.md

```diff
@@ -0,0 +1,123 @@
+
+</aside>
+
+Delete the `cabpk-system` namespace by running:
+
+```bash
+kubectl delete namespace cabpk-system
+```
+
+#### Delete the core and infrastructure provider controller-manager deployments
+
+Delete the `capi-controller-manager` deployment from the `capi-system` namespace:
+
+```bash
+kubectl delete deployment capi-controller-manager -n capi-system
+```
+
+Depending on your infrastructure provider, delete the controller-manager deployment.
+
+For example, if you are using the [AWS provider], delete the `capa-controller-manager` deployment from the `capa-system` namespace:
+
+```bash
+kubectl delete deployment capa-controller-manager -n capa-system
+```
+
+#### Optional: Ensure preserveUnknownFields is set to 'false' for the infrastructure provider CRDs Spec
+This should be the case for all infrastructure providers using conversion webhooks to allow upgrading from `v1alpha2` to
+`v1alpha3`.
+
+This can verified by running `kubectl get crd <crd name>.infrastructure.cluster.x-k8s.io -o yaml` for all the
+infrastructure provider CRDs.
+
+### Upgrade Cluster API components using clusterctl
+
+Run [clusterctl init] with the relevant infrastructure flag. For the [AWS provider] you would run:
+
+```bash
+clusterctl init --infrastructure aws
+```
+
+You should now be able to manage your resources using the `v1alpha3` version of the Cluster API components.
+
+### Adopting existing machines into KubeadmControlPlane management
+
+<aside class="note warning">
+
+<h1> Important </h1>
+
+You must be running at least Cluster API 0.3.7 for adoption to be successful.
+
+</aside>
+
+If your cluster has existing machines labeled with `cluster.x-k8s.io/control-plane`, you may opt in to management of those machines by
+creating a new KubeadmControlPlane object and updating the associated Cluster object's `controlPlaneRef` like so:
+
+```
+---
+apiVersion: "cluster.x-k8s.io/v1alpha3"
+kind: Cluster
+...
+spec:
+  controlPlaneRef:
+    apiVersion: controlplane.cluster.x-k8s.io/v1alpha3
+    kind: KubeadmControlPlane
+    name: controlplane
+    namespace: default
+...
+```
+
+Caveats:
+
+* The KCP controller will refuse to adopt any control plane Machines not bootstrapped with the kubeadm bootstrapper.
+* The KCP controller may immediately begin upgrading Machines post-adoption if they're out of date.
+* The KCP controller attempts to behave intelligently when adopting existing Machines, but because the bootstrapping process sets various fields in the KubeadmConfig of a machine it's not always obvious the original user-supplied `KubeadmConfig` would have been for that machine. The controller attempts to guess this intent to not replace Machines unnecessarily, so if it guesses wrongly, the consequence is that the KCP controller will effect an "upgrade" to its current config. For full details, see SemanticMerge in the kubeadm bootstrapper's api/equality package.
```

Comment:
AFAIK we are not using semantic merge anymore for KCP (In case I recently update the KCP proposal)

Reply:
Yeah, this is very old text that I moved here without fact checking. Probably better if I just link to the proposal.

Reply:
Just checked. KCP proposal don't really describe in detail. I've removed "For full details, see SemanticMerge in the kubeadm bootstrapper's api/equality package.", although we are doing a semantic merge of sorts.